src/shared/dissect-image.c

```diff
@@ -326,6 +327,16 @@ static int device_monitor_handler(sd_device_monitor *monitor, sd_device *device,
         return sd_event_exit(sd_device_monitor_get_event(monitor), r);
 }
 
+static int timeout_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        struct wait_data *w = userdata;
+        int r;
+
+        assert(w);
+
```

Comment:
could you add a comment here explaining what we do here? maybe even a debug message that says explicitly: "Device appeared after timeout reached" or so. And maybe another one that says "Device still not appeared after timeout".